### PR TITLE
23 ALA export to GBIF

### PR DESCRIPTION
**Pull request by muttcg**
This is a draft PR. I checked several ALA datasets via GBIF "download DWCA" link and found that there are only 40 terms inside, so I implemented converter for those fields first and I have the following questions:

1) What terms do I need, all these? - https://github.com/AtlasOfLivingAustralia/biocache-store/blob/develop/src/main/scala/au/org/ala/biocache/export/DwCAExporter.scala#L34
2) Term DwcTerm.class_ contains raw class value, is it a bug in **IndexRecordTransform**?
3) I didn't find interpreted DwcTerm.eventDate, only raw_eventDate, is it ok?
4) For multivalued fields I use **;** as a separator, is that ok?

### CODE REVIEW

**Comment by javier-molina** [comment]
@charvolant, @sadeghim could you provide an insight to Nikolai questions?

I will ping Peggy via Slack I could not tag her here.

**Comment by djtfmartin** [comment]
thanks @muttcg  - this is a big help

1. Yes  - the list here is a good start DwCAExporter.scala

2. _Term DwcTerm.class_ contains raw class value, is it a bug in IndexRecordTransform?_ - Ill have a look into that...

3. _I didn't find interpreted DwcTerm.eventDate, only raw_eventDate, is it ok?_ --We are indexing`eventDate` - it is in our generated index in UAT (with 47m records having a value), but ill review

4. _For multivalued fields I use ; as a separator, is that ok?_ - if this works for GBIF, then works for us 👍

**Comment by timrobertson100** [comment]
From Darwin Core:

> Recommended best practice is to separate the values in a list with space vertical bar space ( | ).

**Comment by peggynewman** [comment]
Hi there, sorry I missed this. I was looking for how to ask to make (1) happen, and include as many DwC terms as possible in the export.

**Comment by muttcg** [comment]
> Hi there, sorry I missed this. I was looking for how to ask to make (1) happen, and include as many DwC terms as possible in the export.

Hi! 
Terms mapping is here [CoreCsvConverter.java](https://github.com/gbif/pipelines/blob/la_23_ala_export_to_gbif/livingatlas/pipelines/src/main/java/au/org/ala/pipelines/converters/CoreCsvConverter.java#L19) it is a copy of [biocache DwCAExporter](https://github.com/AtlasOfLivingAustralia/biocache-store/blob/develop/src/main/scala/au/org/ala/biocache/export/DwCAExporter.scala#L34) and maps 183 terms, here is some tests [CoreCsvConverterTest](https://github.com/gbif/pipelines/blob/la_23_ala_export_to_gbif/livingatlas/pipelines/src/test/java/au/org/ala/pipelines/converters/CoreCsvConverterTest.java#L40)
When you run [IndexRecordToDwcaPipeline](https://github.com/gbif/pipelines/blob/la_23_ala_export_to_gbif/livingatlas/pipelines/src/main/java/au/org/ala/pipelines/beam/IndexRecordToDwcaPipeline.java), it produces 3 files (occurrence.csv, image.csv, and meta.xml), you need to provide eml.xml and compress all 4 files after.

You need any help with the launch of the pipeline, please let me know.

**Comment by djtfmartin** [comment]
I've added the EML to the export

File: livingatlas/pipelines/pom.xml [hunk 1]
```diff
@@ -895,5 +895,10 @@
         <artifactId>checker-qual</artifactId>
         <version>${checker-qual.version}</version>
     </dependency>
+
+    <dependency>
```
Comment (charvolant) [inline livingatlas/pipelines/pom.xml:899]: This dependency seems to also be in the parent POM

**Review by charvolant — APPROVED** [review]

**Pull request merged**